release/src/main/groovy/run_release_candidate_python_validation.sh

```diff
@@ -0,0 +1,272 @@
+    . ./google-cloud-sdk/path.bash.inc
+    gcloud components update --quiet || echo 'gcloud components update failed'
+    gcloud -v
+}
+
+complete() {
+    print_separator "Validation $1"
+    rm -rf $TMPDIR
+}
+
+run_pubsub_publish(){
+    words=("hello world!", "I like cats!", "Python", "hello Python", "hello Python")
+    for word in ${words[@]}
+    do
+        gcloud pubsub topics publish $PUBSUB_TOPIC1 --message "$word"
+    done
+    sleep 10
+}
+
+run_pubsub_pull(){
+    gcloud pubsub subscriptions pull --project=$STREAMING_PROJECT_ID $PUBSUB_SUBSCRIPTION --limit=100 --auto-ack
+}
+
+create_pubsub(){
+    gcloud pubsub topics create --project=$STREAMING_PROJECT_ID $PUBSUB_TOPIC1
+    gcloud pubsub topics create --project=$STREAMING_PROJECT_ID $PUBSUB_TOPIC2
+    gcloud pubsub subscriptions create --project=$STREAMING_PROJECT_ID $PUBSUB_SUBSCRIPTION --topic $PUBSUB_TOPIC2
+}
+
+cleanup_pubsub(){
+    gcloud pubsub topics delete --project=$STREAMING_PROJECT_ID $PUBSUB_TOPIC1
+    gcloud pubsub topics delete --project=$STREAMING_PROJECT_ID $PUBSUB_TOPIC2
+    gcloud pubsub subscriptions delete --project=$STREAMING_PROJECT_ID $PUBSUB_SUBSCRIPTION
+}
+
+
+# Python Release Candidate Configuration
+VERSION="2.3.0"
+CANDIDATE_URL="https://dist.apache.org/repos/dist/dev/beam/$VERSION/"
+SHA1_FILE_NAME="apache-beam-$VERSION-python.zip.sha1"
+MD5_FILE_NAME="apache-beam-$VERSION-python.zip.md5"
+BEAM_PYTHON_SDK="apache-beam-$VERSION-python.zip"
+BEAM_PYTHON_RELEASE="apache-beam-$VERSION-source-release.zip"
+
+# Cloud Configurations
+PROJECT_ID='apache-beam-testing'
+BUCKET_NAME='temp-storage-for-release-validation-tests'
+TEMP_DIR='/quickstart'
+STREAMING_PROJECT_ID='apache-beam-testing'
+STREAMING_BUCKET_NAME='temp-storage-for-release-validation-tests'
+STREAMING_TEMP_DIR='/quickstart'
+NUM_WORKERS=1
+WORDCOUNT_OUTPUT='wordcount_direct.txt'
+PUBSUB_TOPIC1='wordstream-python-topic-1'
+PUBSUB_TOPIC2='wordstream-python-topic-2'
+PUBSUB_SUBSCRIPTION='wordstream-python-sub2'
+
+TMPDIR=$(mktemp -d)
+echo $TMPDIR
+pushd $TMPDIR
+
+#
+# 1. Download files from RC staging location
+#
+wget $CANDIDATE_URL$SHA1_FILE_NAME
+wget $CANDIDATE_URL$MD5_FILE_NAME
+wget $CANDIDATE_URL$BEAM_PYTHON_SDK
+wget $CANDIDATE_URL$BEAM_PYTHON_RELEASE
+
+
+#
+# 2. Verify sha1 and md5 hashes
+#
+
+print_separator "Checking sha1 and md5 hashes"
+hash_check=$(sha1sum -c $SHA1_FILE_NAME | head -1 |awk '{print $2}')
+if [[ "$hash_check" != "OK" ]]
+then
+  echo "ERROR: The sha1 hash doesn't match."
+  complete "The sha1 hash doesn't match."
+  exit 1
+fi
+hash_check=$(md5sum -c $MD5_FILE_NAME | head -1 |awk '{print $2}')
+if [[ "$hash_check" != "OK" ]]
+then
+  echo "ERROR: The md5 hash doesn't match."
+  complete "The md5 hash doesn't match."
+  exit 1
+fi
+echo "SUCCEED: Hashes verification completed."
+
+#
+# 3. create a new virtualenv and install the SDK
+#
+
+print_separator "Creating new virtualenv and installing the SDK"
+
+unzip $BEAM_PYTHON_RELEASE
```

Comment:
There is no need to unzip the source file. What you need is (after creating the virtual env):
pip install $BEAM_PYTHON_SDK[gcp]

and further down when running pipelines use $BEAM_PYTHON_SDK as the sdk_location.

(Unless you want to verify something in the source distribution. That is typically extra for release verification.)